lib/src/main/java/io/ably/lib/util/HttpCodes.java

```diff
@@ -0,0 +1,86 @@
+package io.ably.lib.util;
+
+public enum HttpCodes {
```

Comment:
@qsdigor I strongly feel, it's better to use static class instead of complex enum. So, replacement will be 
```
    public static class HttpCode {
        public static final int  CONTINUE = 100;
        public static final int  SWITCHING_PROTOCOLS= 101;
        ......
    }  
```
So, while accessing, you just need to do 
```
System.out.println(HttpCode.OK) // This is more readable
```
instead of 
```
System.out.println(OK.code) // Not readable
```

I don't feel the message field specified in enum class `HttpCodes`  provides any special meaning, rather adds duplication for the same string constant.

Reply:
Enums are not much flexible in Java. So, most of the time, they do not fit in the expected use case.

Reply:
Agreed, I will implement your suggestions.

Reply:
Looks good and readable now : )